keras/engine/training.py

```diff
@@ -2354,6 +2366,12 @@ def load_weights(self,
         else:
           hdf5_format.load_weights_from_hdf5_group(f, self.layers)
 
+      weight_names = hdf5_format.load_attributes_from_hdf5_group(f, 'weight_names')
+      weight_values = [np.asarray(f[weight_name]) for weight_name in weight_names]
+      weights = self._trainable_weights + self._non_trainable_weights
+      weight_value_tuples = zip(weights, weight_values)
+      backend.batch_set_value(weight_value_tuples)
```

Comment:
Likewise, consider abstracting this code block so we can reuse it in both locations.